src/api/general-services/pipeline-manage/index.js

```diff
@@ -244,13 +243,35 @@ const createPipeline = async (experimentId, processingConfigUpdates) => {
     });
   }
 
+
+  // This is the processing configuration merged for multiple samples where
+  // appropriate.
+  // eslint-disable-next-line consistent-return
+  const mergedProcessingConfig = _.cloneDeepWith(processingConfig, (o) => {
+    if (_.isObject(o) && !o.dataIntegration && o.enabled) {
+      // Find which samples have sample-specific configurations.
+      const sampleConfigs = _.intersection(Object.keys(o), samples.ids);
+
+      // Get an object that is only the "raw" configuration.
+      const rawConfig = _.omit(o, sampleConfigs);
+
+      const result = {};
+
+      samples.ids.forEach((sample) => {
+        result[sample] = _.merge(rawConfig, o[sample]);
+      });
+
+      return result;
+    }
+  });
+
```

Comment:
This takes a processing config like:

```
{
      processingConfig: {
        cellSizeDistribution: {
          filterSettings: { binStep: 200, minCellSize: 420 }, enabled: true,
        },
      },
    }
```

and turns it into something like:

```
{
  processingConfig: {
    cellSizeDistribution: {
      'sample-control': {
        filterSettings: { binStep: 200, minCellSize: 420 }, enabled: true,
      },
      'sample-treated': {
        filterSettings: { binStep: 200, minCellSize: 420 }, enabled: true,
      },
    },
  },
}
```

Where each object is a fully valid config on its own. It will also take any partial config that may be submitted specifically for a sample (sample-specific configs need not contain all required fields) and merges them into the "root" config file to create a fully valid processing config for each sample.

Reply:
Would it make sense to handle enabled/disabled on the api side? By perhaps generating the skeleton for the steps according to the enabled/disabled steps?